test/unit/helpers/utils.ts

```diff
@@ -0,0 +1,7 @@
+export function onPage(name: string) {
+  window.history.replaceState(null, `Testing Page: ${name}`, `${name}.html`)
+}
+
+export function flushPromises() {
```

Comment:
I see this used in the test you provided, would you mind explaining why we need it?

Reply:
Yeah, it's for when there are pending promises out there in the app that can't be awaited here in the test. This says, wait for all the pending promises to be fulfilled in the next event loop. 

This happens with our mock-ipc calls. A promise is returned when we invoke something, then the next tick, the promise is fulfilled. This test needs to wait for those ipc channels to send their replies.